backend/module/chaos/serverexperimentation/xds/xds_cache.go

```diff
@@ -77,37 +82,56 @@ func refreshCache(ctx context.Context, storer experimentstore.Storer, snapshotCa
 
 	// Create/Update experiments
 	for cluster, experiments := range clusterFaultMap {
+		resources := make(map[gcpTypes.ResponseType][]gcpTypes.ResourceWithTtl)
+
 		logger.Debugw("Injecting fault for cluster", "cluster", cluster)
 
-		runtimeResource := generateRTDSResource(experiments, rtdsConfig, ttl, logger)
-		err := setSnapshot(runtimeResource, cluster, snapshotCache, logger)
+		// Enable ECDS if cluster present in the map
+		if _, exists := ecdsConfig.enabledClusters[cluster]; exists {
+			resources[gcpTypes.ExtensionConfig] = generateECDSResource(experiments, cluster, ttl, logger)
+
+			// generate empty Runtime config to clear any existing RTDS faults
```

Comment:
This is also to provide an initial RTDS response to clients booting up right? Maybe add to comment or be less specific